packages/expansion-hub/src/internal/ExpansionHub.ts

```diff
@@ -431,8 +597,18 @@ export class ExpansionHubInternal implements ExpansionHub {
 
     async addChildByAddress(moduleAddress: number): Promise<RevHub> {
         let childHub = new ExpansionHubInternal(false, this.serialPort);
-        await childHub.open(moduleAddress);
-        await childHub.queryInterface("DEKA");
+
+        try {
+            await childHub.open(moduleAddress);
+            await childHub.queryInterface("DEKA");
+        } catch (e: any) {
+            //errorCode = -2 indicates timeout
```

Comment:
If -2 is timeout, why doesn't createError() handle that? Are there other error codes that it doesn't handle?

Reply:
In this case, we assume that a timeout means there is no device connected (since this is the first time we try to open it). In other cases, a timeout means we lost connection. It may be good to create a TimeoutError and have this catch block check if the error is an instance of TimeoutError.